static/app/components/compactSelect/composite.tsx

```diff
@@ -0,0 +1,208 @@
+import {Children, useMemo} from 'react';
+import styled from '@emotion/styled';
+import {FocusScope} from '@react-aria/focus';
+import {Item} from '@react-stately/collections';
+
+import space from 'sentry/styles/space';
+
+import {Control, ControlProps} from './control';
+import {ListBox, MultipleListBoxProps, SingleListBoxProps} from './listBox';
+import {SelectOption} from './types';
+
+interface BaseCompositeSelectRegion<Value extends React.Key> {
```

Comment:
Should the value just be the key since we're constraining value to key anyway?

Reply:
Then we don't have to specify both

Reply:
Are you referring to the `key` property (`key: React.Value`) or the fact that `BaseCompositeSelectRegion` is a generic type?

Reply:
So `Value extends React.Key` means that Value MUST be a subtype of `React.Key`, and since `key` is already of type `React.Key` we could make this `key?: Value` and then just prefill it from value.

Reply:
or maybe that's what this is already doing actually now that I think about it

Reply:
So `Value` is a generic parameter that can be something more specific than `React.Key`:
<img width="1094" alt="Screenshot 2023-01-20 at 3 22 47 PM" src="https://user-images.githubusercontent.com/44172267/213822691-a2e3fe89-a950-4045-80a8-d772dee55c4c.png">

I don't think we want to force the `key` property to be of type `Value` in such cases. `key` is only meant to help distinguish between composite regions – it's not a value type.